bundles/org.openhab.binding.solarmax/src/main/java/org/openhab/binding/solarmax/internal/SolarMaxHandler.java

```diff
@@ -0,0 +1,154 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolarMaxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jamie Townsend - Initial contribution
+ */
+public class SolarMaxHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SolarMaxHandler.class);
+
+    @Nullable
+    private SolarMaxConfiguration config;
+
+    @Nullable
+    private ScheduledFuture<?> pollingJob;
+
+    public SolarMaxHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // Read only
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing SolarMax");
+
+        config = getConfigAs(SolarMaxConfiguration.class);
+
+        configurePolling(); // Setup the scheduler
+    }
+
+    /**
+     * This is called to start the refresh job and also to reset that refresh job when a config change is done.
+     */
+    private void configurePolling() {
+        logger.debug("Polling data from {} at {}:{} every {} seconds ", getThing().getUID(), this.config.host,
+                this.config.portNumber, this.config.refreshInterval);
+        if (this.config.refreshInterval > 0) {
+            if (pollingJob == null || pollingJob.isCancelled()) {
+                pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, this.config.refreshInterval,
+                        TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing SolarMax Handler Thing");
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+        }
+        pollingJob = null;
+    }
+
+    /**
+     * Polling event used to get data from the SolarMax device
+     */
+    private Runnable pollingRunnable = () -> {
+        updateValuesFromDevice();
+    };
+
+    private synchronized void updateValuesFromDevice() {
+
+        logger.debug("Updating data from {} at {}:{} ", getThing().getUID(), this.config.host, this.config.portNumber);
+        // get the data from the SolarMax device
+        try {
+            SolarMaxData solarMaxData = SolarMaxConnector.getAllValuesFromSolarMax(config.host, config.portNumber);
+
+            if (solarMaxData.wasCommunicationSuccessful()) {
+                updateChannels(solarMaxData);
+                updateStatus(ThingStatus.ONLINE);
+                return;
+            }
+        } catch (SolarMaxException e) {
+            logger.debug("Error refreshing source {} at {}:{} - {}", getThing().getUID(), this.config.host,
+                    this.config.portNumber, e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Communication error with the device. Please retry later.");
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

You could include the exceptions message as a third parameter in `updateStatus` to give the user a clue, what's wrong.